pkg/provider/googleapps/googleapps.go

```diff
@@ -51,52 +51,102 @@ func (kc *Client) Authenticate(loginDetails *creds.LoginDetails) (string, error)
 
 	authForm.Set("Email", loginDetails.Username)
 
-	passwordURL, _, err := kc.loadLoginPage(authURL+"?hl=en&loc=US", loginDetails.URL+"&hl=en&loc=US", authForm)
+	passwordURL, passwordForm, err := kc.loadLoginPage(authURL+"?hl=en&loc=US", loginDetails.URL+"&hl=en&loc=US", authForm)
 	if err != nil {
 		return "", errors.Wrap(err, "error loading login page")
 	}
 
 	logger.Debugf("loginURL: %s", passwordURL)
 
 	authForm.Set("Passwd", loginDetails.Password)
-	authForm.Set("rawidentifier", loginDetails.Username)
 
-	responseDoc, err := kc.loadChallengePage(passwordURL+"?hl=en&loc=US", authURL, authForm, loginDetails)
+	referingURL := passwordURL
+
+	if _, rawIdPresent := passwordForm["rawidentifier"]; rawIdPresent {
```

Comment:
I suggest using the variable name `rawIDPresent`.
https://github.com/golang/go/wiki/CodeReviewComments#initialisms